src/components/FeatureViews/Accounts/TabComponents/StudentInfo.js

```diff
@@ -1,55 +1,92 @@
-import "../Accounts.scss";
-import * as hooks from "actions/hooks";
+import {Link, useParams} from "react-router-dom";
+import React, {useMemo} from "react";
+import gql from "graphql-tag";
 import Grid from "@material-ui/core/Grid";
 import Loading from "components/OmouComponents/Loading";
-import ProfileCard from "../ProfileCard";
-import PropTypes from "prop-types";
-import React from "react";
-import {useSelector} from "react-redux";
 import LoadingError from "./LoadingCourseError";
+import {makeStyles} from "@material-ui/core/styles";
+import ProfileCard from "../ProfileCard";
+import {useQuery} from "@apollo/react-hooks";
 
-const StudentInfo = ({user}) => {
-	const studentList = useSelector(({Users}) => Users.StudentList);
-	const fetchStatus = hooks.useStudent(user.student_ids);
-	const loadedStudentIDs = user.student_ids.filter(
-		(studentID) => studentList[studentID]
-	);
-
-	if (loadedStudentIDs.length === 0) {
-		if (hooks.isLoading(fetchStatus)) {
-			return <Loading/>;
-		}
-		if (hooks.isFail(fetchStatus)) {
-			return <LoadingError error="students"/>;
-		}
-	}
-
-	return (
-		<Grid
-			alignItems="center"
-			container
-			direction="row"
-			md={12}
-			spacing={5}
-			xs={10}
-		>
-			{loadedStudentIDs.map((studentID) => (
-				<ProfileCard
-					key={studentID}
-					route={`/accounts/student/${studentID}`}
-					user={studentList[studentID]}
-				/>
-			))}
-		</Grid>
-	);
-};
+const useStyles = makeStyles({
+    "center": {
+        "margin": "auto",
+    },
+    "new": {
+        "backgroundColor": "#f5f5f5",
+        "border": "1.5px dashed #999999",
+        "cursor": "pointer",
+        "height": "50px",
+        "position": "relative",
+    },
+});
+
+
+const GET_STUDENTS = gql`
+    query GetStudents($id: ID) {
+        parent(userId: $id) {
+            studentPrimaryParent {
+                ...StudentInfo
+            }
+            studentSecondaryParent {
+                ...StudentInfo
+            }
+        }
+    }
+
+    fragment StudentInfo on StudentType {
+        phoneNumber
+        user {
+            id
+            firstName
+            lastName
+            email
+        }
+    }`;
 
-StudentInfo.propTypes = {
-	user: PropTypes.shape({
-		student_ids: PropTypes.arrayOf(
-			PropTypes.oneOfType([PropTypes.string, PropTypes.number])
-		).isRequired,
-	}).isRequired,
+const StudentInfo = () => {
+    const {accountID} = useParams();
+    const {data, loading, error} = useQuery(GET_STUDENTS, {
+        "variables": {"id": accountID},
+    });
+
+    const classes = useStyles();
+
```

Comment:
Do additional access control here to verify the current user is either the student's parent or is a receptionist or admin user. If the user isn't a parent or a receptionist, be sure they can't access the `studentInvite` link (aka, set `studentInvite` to false when it's not one of these 2)